Std/Data/BitVec/Lemmas.lean

```diff
@@ -89,6 +89,9 @@ theorem eq_of_toFin_eq {x y : BitVec w} (w : x.toFin = y.toFin) : x = y := by
 theorem ofNat_one (n : Nat) : BitVec.ofNat 1 n = BitVec.ofBool (n % 2 = 1) :=  by
   rcases (Nat.mod_two_eq_zero_or_one n) with h | h <;> simp [h, BitVec.ofNat, Fin.ofNat']
 
+theorem ofBool_eq_iff_eq (b b' : Bool) : BitVec.ofBool b = BitVec.ofBool b' ↔ b = b' := by
+  constructor <;> intros h <;> cases b <;> cases b' <;> simp at h ⊢ <;> try contradiction
```

Comment:
You don't need the `constructor` tactic, `simp` is able to simplify trivial `iff`s pretty well. I also squeezed the non-terminal simp, negating some of the golfing:
```suggestion
  cases b <;> cases b'
    <;> simp only [ofBool_false, ofNat_eq_ofNat, ofBool_true, iff_false]
    <;> (intro; contradiction)
```